legion/engine/physics/broadphase_collision_algorithms/broadphase_uniformgrid.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <physics/broadphase_collision_algorithms/broadphasecollisionalgorithm.hpp>
+#include <physics/physics_statics.hpp>
+
+namespace legion::physics
+{
+
+    class BroadphaseUniformGrid : public BroadPhaseCollisionAlgorithm
+    {
+    public:
+        BroadphaseUniformGrid(math::ivec3 cellSize) : m_cellSize(cellSize)
+        {
+        }
+
+        void collectPairs(std::vector<physics_manifold_precursor>& manifoldPrecursors,
+            std::vector<std::vector<physics_manifold_precursor>>& manifoldPrecursorGrouping) override;
+
+        void setCellSize(math::ivec3 cellSize)
+        {
+            m_cellSize = cellSize;
+        }
+
+    private:
+        math::ivec3 m_cellSize;
+
+        math::ivec3 calculateCellIndex(const math::vec3 point);
+    };
+}
```

Comment:
this could really use some comments, explain the strategy, and what the benefits over brute force are.
Use a `@class`-doxy header